react/features/base/ui/components/BaseTheme.web.js

```diff
@@ -1,7 +1,93 @@
 // @flow
 
-import { font, colors, colorMap, spacing, shape, typography } from '../Tokens';
-import { createWebTheme } from '../functions';
+import { createMuiTheme } from '@material-ui/core/styles';
+
+import { font, colors, colorMap, spacing, shape, typography, breakpoints } from '../Tokens';
+import { createWebTheme, overwriteRecurrsive } from '../functions';
+import { createColorTokens } from '../utils';
+
+/**
+ * Cached branded theme, in order to avoid re-computation of branded theme on re-renders.
+ */
+let brandedTheme;
+
+/**
+ * Applies branding to theme, if any.
+ *
+ * @param {Function|Object} stateful - The redux store or {@code getState} function.
+ * @returns {Object} - The processed theme.
+ */
+export function applyBrandingTheme(stateful: Function | Object) {
+    const { customTheme } = stateful['features/dynamic-branding'];
+
+    if (!customTheme) {
+        return;
+    }
+
+    if (brandedTheme) {
+        return brandedTheme;
```

Comment:
I think it will be better to have the new branded theme in the store instead of caching it here. This way we could easily access it.

Reply:
excelent idea!